jobbergate-api/jobbergate_api/apps/applications/routers.py

```diff
@@ -293,14 +283,19 @@ async def applications_get_by_id(application_id: int = Query(...)):
             detail=message,
         )
 
-    logger.debug(f"Application data: {application_data=}")
-
-    response = ApplicationResponse(**application_data)
+    logger.trace(f"Application data: {dict(application_data)}")
 
-    if response.application_uploaded:
-        application_files = get_application_files_from_s3(response.id)
-        response.application_source_file = application_files.source_file
-        response.application_templates = application_files.templates
+    if application_data["application_uploaded"]:
+        response = ApplicationResponse(
+            **application_data,
+            **ApplicationFiles.get_from_s3(application_data["id"]).dict(
```

Comment:
Since application files are optional (i.e., they can be uploaded/deleted and the application will be fine), `get_from_s3` does not fail. But I see that if the application is marked in the database as uploaded, we should be sure that `get_from_s3` found the files.